### ISSUE TITLE
Option to disable persistent shared Windows file notification

### ISSUE THREAD

**Issue by phillip-elm** [docker/for-win#7555]
In the Edge version of Docker for Windows, the following nag appears every time a container is started that shares Windows files.

![](https://i.imgur.com/0jJ50AP.png)

Despite the hit in performance, my use case requires that I have the files on the host filesystem. In circumstances where containers are being spun up frequently, this notification quickly wears out its welcome.

**That said, it would be great if:**

1. Users could acknowledge and disable this warning henceforth
2. There were a way to do this via configuration or programmatically

### Information
  - Windows Version: Windows 10 2004 WSL2
  - Docker Desktop Version: 2.3.2.1 (edge)
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: No

### Steps to reproduce the behavior

1. Start a docker container with a shared Windows directory / file

**Comment by simonferquel** [comment]
Hi, we received quite a lot of feedback about this, we are going to throttle this so that it appears no more than once a day. Would it make it more useful to you ?

**Comment by phillip-elm** [comment]
Salut @simonferquel!

I'm all for performance - so I appreciate the idea - but unfortunately this does not solve the issue for me.

I'm in a situation where I have Linux and Mac versions of my toolkit with a Windows release in the works (now that WSL2 is GA). For this specific application, the use case requires that the mount be on the host OS, and the performance hit isn't noticeable - so re-engineering a large portion of the application to hide a warning on Windows doesn't make much sense.

For a criticial warning such as potential data loss, instability, or a deprecation (_with notice_) - I would absolutely agree with nagging the user - but this is (or seems to be), at worst, a non-critical issue caused by use of a feature that is commonly used on Linux and Mac.

It would be great to see this warning move to the command line, instead of forcing the warning via OS-level notifs. This way, developers will see the warning, but end users won't be bothered by it. To my knowledge, this is how the majority of other developer tools handle problems like this.

I suspect that as more and more developers port their software to Windows for use with DFW, this notification will get more negative attention - unless it moves to the terminal, or gets an option to permanently hide it.

**tl;dr:** Can the warning be moved to the terminal instead of the GUI?

**Comment by simonferquel** [comment]
Unfortunately, the warning can't move to the terminal, as we don't modify the docker CLI for Docker Desktop.
There is a way to avoid the warning though running something like `docker run -v $(wslpath -u c:\something):/somewhere` from the wsl command line. It has the same performance profile, but does not trigger the notification.

**Comment by phillip-elm** [comment]
How can this workaround be done with Compose or Docker on the Windows command line? In Compose, relative paths are common, and in my tests they do appear to trigger the warning.

This isn't a critical notification or error - if the warning can't be moved to the CLI, then the option to permanently silence it needs to be added.

**Comment by phillip-elm** [comment]
So to answer my own question, starting Docker Compose in Linux from Windows is pretty easy:

```bash
wsl docker-compose up -d
```

On the Ubuntu distro, this starts the container without a hitch, no warning, everyone's happy. 

Unfortunately, I can't replicate this with the Docker-provided distro, "docker-desktop"

Running ```wsl -d docker-desktop docker-compose up -d``` shows:

```
The command 'docker-compose' could not be found in this WSL 2 distro.
We recommend to activate the WSL integration in Docker Desktop settings.

See https://docs.docker.com/docker-for-windows/wsl/ for details.
```

This appears to be an issue outlined in https://github.com/docker/for-win/issues/6876.

This is a lot of hoops to jump through just to hide a warning. :confused:

**Comment by simonferquel** [comment]
We are adding the option to mute the popup. (default will be to show it at most once per 12 hours, with a button in the popup to mute it forever).

**Comment by phillip-elm** [comment]
Awesome, thanks!

**Comment by stephen-turner** [comment]
Released in Edge 2.3.3.0.
